### PR TITLE
updated to rails 7.0.4

### PR DESCRIPTION
**Pull request by aquillo-au**
Have updated from rails 6 to 7,
Attempted to make as small changes as possible while still having everything work.

### CODE REVIEW

**Comment by aquillo-au** [comment]
does that mean i need to fix the rubocop stuff 1st?

File: Gemfile [hunk 1]
```diff
@@ -47,8 +48,8 @@ group :development do
   gem 'listen'
   # Spring speeds up development by keeping your application running in the
   # background. Read more: https://github.com/rails/spring
-  gem 'spring'
-  gem 'spring-watcher-listen'
+  gem 'spring', '~> 4.1'
+ # gem 'spring-watcher-listen'
```
Comment (leesheppard) [inline Gemfile:52]: if this is no longer required or wanted, then remove rather than comment out.

Comment (aquillo-au) [reply Gemfile:52]: will do

**Comment by leesheppard** [comment]
@aquillo-au resolve the build issues and push up again, check the errors locally with Rubocop first to ensure a successful build.

**Pull request closed without merging**